vertical-pod-autoscaler/pkg/admission-controller/Dockerfile

```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM --platform=$BUILDPLATFORM golang:1.22.4 as builder
+FROM --platform=$BUILDPLATFORM golang:1.23rc1 as builder
```

Comment:
As per the following [comment](https://github.com/kubernetes/autoscaler/pull/7002#discussion_r1664703753), I don't think this is correct.

Reply:
cc @voelzmo @kgolab

Reply:
Closing this PR as dependabot opened PR #7008 to update the Golang version to 1.22.5

/close